Lib/http/cookiejar.py

```diff
@@ -1044,12 +1044,13 @@ def set_ok_domain(self, cookie, request):
             else:
                 undotted_domain = domain
             embedded_dots = (undotted_domain.find(".") >= 0)
-            if not embedded_dots and domain != ".local":
+            if not embedded_dots and not erhn.endswith(".local"):
                 _debug("   non-local domain %s contains no embedded dot",
                        domain)
                 return False
             if cookie.version == 0:
-                if (not erhn.endswith(domain) and
+                if (not (erhn.endswith(domain) or
+                         (erhn.endswith(".local") and erhn[:len(erhn)-6].endswith(undotted_domain))) and
```

Comment:
```suggestion
                         erhn.endswith(f"{undotted_domain}.local")) and
```
This is equivalent, right?

Reply:
Yes, it definitely is, I'll commit it